chromeapp/app/scripts/chrome-apis/file-system.js

```diff
@@ -0,0 +1,144 @@
+/* globals Promise, chrome */
+'use strict';
+
+/**
+ * This module provides a wrapper around the callback-heavy chrome.fileSystem
+ * API and provides an alternative based on Promises.
+ */
+
+/**
+ * @param {Entry} entry
+ *
+ * @return {Promise} Promise that resolves with the display path
+ */
+exports.getDisplayPath = function(entry) {
+  return new Promise(function(resolve) {
+    chrome.fileSystem.getDisplayPath(entry, function(displayPath) {
+      resolve(displayPath);
+    });
+  });
+};
+
+/**
+ * @param {Entry} entry the starting entry that will serve as the base for a
+ * writable entry
+ *
+ * @return {Promise} Promise that resolves with a writable entry
+ */
+exports.getWritableEntry = function(entry) {
+  return new Promise(function(resolve) {
+    chrome.fileSystem.getWritableEntry(entry, function(writableEntry) {
+      resolve(writableEntry);
+    });
+  });
+};
+
+/**
+ * @param {Entry} entry
+ *
+ * @return {Promise} Promise that resolves with a boolean
+ */
+exports.isWritableEntry = function(entry) {
+  return new Promise(function(resolve) {
+    chrome.fileSystem.isWritableEntry(entry, function(isWritable) {
+      resolve(isWritable);
+    });
+  });
+};
+
+/**
+ * The original Chrome callback takes two parameters: an entry and an array of
```

Comment:
Can you file a crbug about this? It does not make sense that the documentation doesn't explain what the second parameter is for (and I agree that the docs are confusing).